lab_2_tokenize_by_bpe/main.py

```diff
@@ -47,6 +88,26 @@ def merge_tokens(
     :param pair: a pair of tokens to be merged
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if (not isinstance(word_frequencies, dict) or
+            not isinstance(pair, tuple)):
+        return None
+
+    new_word_freq = word_frequencies.copy()
+    for word in word_frequencies:
+        new_word = []
+        if pair[0] in word and pair[1] in word:
+            for i in range(len(word) - 1):
+                if word[i] == pair[1] and word[i - 1] == pair[0]:
```

Comment:
why do you check by elements? Looks like you can compare full tuples

Reply:
fixed